ts/src/base/Exchange.ts

```diff
@@ -3429,7 +3429,7 @@ export default class Exchange {
 
     costToPrecision (symbol: string, cost) {
         const market = this.market (symbol);
-        return this.decimalToPrecision (cost, TRUNCATE, market['precision']['price'], this.precisionMode, this.paddingMode);
+        return this.decimalToPrecision (cost, TRUNCATE, this.safeNumber2 (market['precision'], 'cost', 'price'), this.precisionMode, this.paddingMode);
```

Comment:
the fun part here is that, if precision is DECIMAL_PLACES/SIGNIFICANT_DIGITS, but user uses `string` for artithmetics, then it would lead to issue, because here : https://github.com/ccxt/ccxt/blob/master/ts/src/base/functions/number.ts#L116

it is treaded as numeric directly